src/test/ui/type-mismatch.stderr

```diff
@@ -194,16 +194,16 @@ LL |     want::<Foo<foo, B>>(f); //~ ERROR mismatched types
    |                         ^ expected struct `B`, found struct `A`
```

Comment:
Note this error becomes awkward:
```
                         ^ expected struct `B`, found struct `A`

    = note: expected type `Foo<_, B>`
              found type `Foo<_, A>`
              found type `Foo<_>`
```

Maybe this particular case should display the parameters.

Reply:
You should probably create a new method similar to `strip_generic_default_params` that checks wether the default associated type has been changed, and if it has, don't call `strip_generic_default_params`. Does that make sense?

This is part of the reason we haven't unified `ppaux` and this: this code is conditional on the state of _two_ types with very specific interactions.

Reply:
Turns out addressing the awkwardness made the change simpler. I added more tests, too.